chapter05/red/exercise2.pl

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/perl
+
+use warnings;
+use strict;
+use List::Util qw( reduce );
+
+my $fn = shift @ARGV // 'coconet.dat';
+open my $fh, '<', $fn;
+
+my %stats;
+while( <$fh> ) {
+  next if /^\s*(#.*)?$/;
```

Comment:
My data file came with empty lines, this regex ensures they're not processed too.